### PR TITLE
Feature/syal 764

### PR DESCRIPTION
**Pull request by MaksimR-AVISPL**

### CODE REVIEW

File: src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+     * allowing to launch one of the available applications.
+     *
+     * @param statistics to keep collected properties
+     * @param controls to keep controllable properties
+     * @throws Exception if any error occurs
+     * */
+    private void generateApplicationsStatistics(Map<String, String> statistics, List<AdvancedControllableProperty> controls) throws Exception {
+        ArrayNode apps = (ArrayNode) this.doPost(APP_CONTROL_URI, getApplicationList, JsonNode.class).at(RESULT_0_URI);
+        String[] appLabels = new String[apps.size() + 1];
+        String[] appOptions = new String[apps.size() + 1];
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java]: no biggie, but `apps.size() + 1` can be extracted and reused.

File: src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+     * Retrieve data from {@link Constant#AUDIO_URI} endpoint with
+     * {@link LCDCommands#getSoundSettings} and {@link LCDCommands#getSpeakerSettings} payloads
+     *
+     * @param statistics to keep collected properties
+     * @param controls to keep controllable properties
+     * @throws Exception if any error occurs
+     * */
+    private void generateAudioStatistics(Map<String, String> statistics, List<AdvancedControllableProperty> controls) throws Exception {
+        ArrayNode soundSettings = (ArrayNode) this.doPost(AUDIO_URI, getSoundSettings, JsonNode.class).at(RESULT_0_URI);
+        ArrayNode speakerSettings = (ArrayNode) this.doPost(AUDIO_URI, getSpeakerSettings, JsonNode.class).at(RESULT_0_URI);
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java:277]: making sure no copy/paste issue here, and both should use `AUDIO_URI` (just pointing it out, and I'm sure it is correct)

File: src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+                        ImmutableMap.of("value", value, "target", SUBWOOFER_FREQ))));
+                break;
+            case SPEAKER_SUBWOOFER_PHASE:
+                this.doPost(AUDIO_URI, setSpeakerSettings.withParams(ImmutableMap.of("settings",
+                        ImmutableMap.of("value", value, "target", SUBWOOFER_PHASE))));
+                break;
+            case SPEAKER_SUBWOOFER_POWER:
+                this.doPost(AUDIO_URI, setSpeakerSettings.withParams(ImmutableMap.of("settings",
+                        ImmutableMap.of("value", "0".equals(value) ? "off" : "on", "target", SUBWOOFER_POWER))));
+                break;
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java:461]: Looks like `AUDIO_URI` is used for any sound related items...ignore my previous comment :)

File: src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+                    errorOutput.append("- ").append(errors.get(i + 1).asText());
+                }
+            }
+            return errorOutput.toString();
+        }
+        return "";
+    }
+
+    @Override
+    public void controlProperty(ControllableProperty cp) throws Exception {
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java:424]: please add `inheritDoc`...all public or protected methods should have javaDocs

File: src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java [hunk 5]
```diff
@@ -0,0 +1,584 @@
...
+                break;
+            default:
+                if (this.logger.isWarnEnabled())
+                    this.logger.warn("Control property \"" + prop + "\" is invalid");
+                throw new UnsupportedOperationException("Control property \"" + prop + "\" is invalid");
+        }
+    }
+
+    @Override
+    public void controlProperties(List<ControllableProperty> list) throws Exception {
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java:498]: please add `inheritDoc`...all public or protected methods should have javaDocs

File: src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java [hunk 6]
```diff
@@ -0,0 +1,584 @@
...
+        this.setBaseUri("sony");
+        this.setContentType("application/json");
+    }
+
+    @Override
+    protected void authenticate() {
+    }
+
+    @Override
+    public List<Statistics> getMultipleStatistics() throws Exception {
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java:53]: please add `inheritDoc`...all public or protected methods should have javaDocs

File: src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java [hunk 7]
```diff
@@ -0,0 +1,584 @@
...
+ * @see <a href="https://pro-bravia.sony.net/develop/integrate/rest-api/spec/">Sony Rest API Spec</a>
+ */
+public class SonyBraviaCommunicator extends RestCommunicator implements Controller, Monitorable {
+    public SonyBraviaCommunicator() {
+        this.setBaseUri("sony");
+        this.setContentType("application/json");
+    }
+
+    @Override
+    protected void authenticate() {
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java:46]: please add `inheritDoc`...all public or protected methods should have javaDocs

File: src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java [hunk 8]
```diff
@@ -0,0 +1,584 @@
...
+    }
+
+    @Override
+    public void controlProperties(List<ControllableProperty> list) throws Exception {
+        for (ControllableProperty cp : list)
+            controlProperty(cp);
+    }
+
+    @Override
+    protected HttpHeaders putExtraRequestHeaders(HttpMethod httpMethod, String uri, HttpHeaders headers) {
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java:507]: please add `inheritDoc`...all public or protected methods should have javaDocs

**Review by ShervinM-AVISPL — APPROVED** [review]
Looks good, approving, but please see few comments that I added before you merge.

File: src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java (hunk 2 above)
Comment (MaksimR-AVISPL) [reply src/main/java/com/insightsystems/dal/sony/SonyBraviaCommunicator.java:277]: yes, it is the same endpoint, different payloads though

**Pull request merged**